### PR TITLE
Merging wp-offline-shell

### PR DESCRIPTION
**Pull request by darkwing**
ToDo:

- [x] - Make tests pass
- [x] - Give plugin its own side menu
- [x] - Complete rebranding of readme, etc.
- [x] - Make SW for assets work.
- [x] - Make shell uninstall work
- [x] - Make shell admin save work
- [x] - Make admin links work in nags
- [x] - Audit plugin for PHP notices, warnings
- [x] - Make shell install/activate/update work

### CODE REVIEW

**Comment by darkwing** [comment]
This is ready for a look when you get back, @delapuente !

File: tests/service-worker/testShell.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
+describe('tests', function() {
+  'use strict';
+
+  var storageKey = 'testCache';
+  var fakeEvent;
+
+  beforeEach(function(done) {
+    // Needed to prevent SW from bombing on what needs to be replaced
```
Comment (delapuente) [inline tests/service-worker/testShell.js]: What is the issue here? I can not to figure out.

File: wp-offline-content/class-wp-offline-content-admin.php [hunk 2]
```diff
@@ -17,6 +17,8 @@ public static function init() {
         return self::$instance;
     }
 
+    public $smo;
```
Comment (delapuente) [inline wp-offline-content/class-wp-offline-content-admin.php]: What does this name stand for? Do you mind to clarify?

File: tests/service-worker/testShell.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
+describe('tests', function() {
```
Comment (delapuente) [inline tests/service-worker/testShell.js]: You forgot to add this test to `sw-tests.js` file. Don't worry, it is working well but please, add it. :)

File: tests/service-worker/testShell.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+              done();
+            });
+          });
+        });
+      });
+    });
+  });
+
+  it('A URL manually removed from cache after it\'s been cached is re-downloaded and cached', function() {
+    // https://github.com/mozilla/offline-shell/issues/43
```
Comment (delapuente) [inline tests/service-worker/testShell.js]: This remind me about moving the issues open in your repo to mine.

File: wp-offline-content/readme.txt [hunk 5]
```diff
@@ -37,7 +37,15 @@ The plugin will try to always serve fresh content from the Internet. After visit
...
+
+= Does this plugin modify my theme files? =
+
+Absolutely not -- no files are modified or generated by this plugin.
+
+= When should the theme file options be updated? =
+
+It's best to update the plugin file options when you've added a new file to your theme that you'd like cached.  Files that change or are deleted will be automatically detected and will not create problems within the service worker.
 
 More options will be available with new versions of the plugin.
```
Comment (delapuente) [inline wp-offline-content/readme.txt]: You should modify the change log as well. This is a huge and exciting addition!

File: wp-offline-content/wp-offline-content.php [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <?php
 /*
-Plugin Name: Offline Content
+Plugin Name: Offline
 Description: Allow your users to read your content even while offline.
 Plugin URI: https://github.com/delapuente/wp-offline-content
 Version: 0.5.0
```
Comment (delapuente) [inline wp-offline-content/wp-offline-content.php]: You should bump the version here.

File: wp-offline-content/wp-offline-content.php [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <?php
 /*
-Plugin Name: Offline Content
+Plugin Name: Offline
```
Comment (delapuente) [inline wp-offline-content/wp-offline-content.php]: Are you sure you want to change the name of the plugin?

File: wp-offline-content/class-wp-offline-content-admin.php [hunk 8]
```diff
@@ -78,16 +80,35 @@ function () {},
     }
 
     public function admin_menu() {
-        add_options_page(
-            __('Offline Content Options', 'offline-content'), __('Offline Content', 'offline-content'),
-            'manage_options', self::$options_page_id, array($this, 'create_admin_page')
-        );
+
+      add_menu_page(
+        'Offline',
```
Comment (delapuente) [inline wp-offline-content/class-wp-offline-content-admin.php]: We should translate this and the title in the next line.

File: wp-offline-content/class-wp-offline-content-admin.php [hunk 9]
```diff
@@ -78,16 +80,35 @@ function () {},
...
+        'Offline',
+        'manage_options',
+        'offline-content',
+        array($this, 'create_content_admin_page')
+      );
+
+      $this->smo = add_submenu_page(
+        'offline-content',
+        'Design',
```
Comment (delapuente) [inline wp-offline-content/class-wp-offline-content-admin.php]: Same here.

**Comment by delapuente** [comment]
Awesome work. I like how the merge is looking. After this, we should homogenize the architecture for the plugin and merge the sw. With that done, we will be ready for tagging version 1.0.0. What do you think?

**Comment by delapuente** [comment]
When entering `Design` option pane, WordPress is showing the following warning:
```
Deprecated: Non-static method Offline_Shell_Admin::options() should not be called statically, assuming $this from incompatible context in /Users/salva/workspace/wp-offline-content/wp-offline-content/class-wp-offline-content-admin.php on line 109
```
Can we try to solve it before merging?

**Comment by delapuente** [comment]
Testing it I've noticed the default theme is still not working as we are not suggesting cache `jQuery`. We should add this kind of files as lot of themes could be requiring them.

**Comment by delapuente** [comment]
Looking at how both plugins fit together, I was wondering if we could get a more appealing name converting the new menu entry from `Offline` to `App Shell`. I'm almost sure the submenus should be named something like: `Content` and `Theme` being WordPress-centric or `Content` and `Shell` being PWA-centric (although current names could be OK as well) but for the top menu I have my doubts.

What do you think @digitarald @marco-c @mykmelez @darkwing ?

captura de pantalla 2016-04-01 a las 13 43 35

File: tests/service-worker/testShell.js (hunk 1 above)
Comment (darkwing) [inline tests/service-worker/testShell.js]: Updating comment.

File: wp-offline-content/class-wp-offline-content-admin.php (hunk 2 above)
Comment (darkwing) [inline wp-offline-content/class-wp-offline-content-admin.php]: Turns out I don't need this, removing.

File: tests/service-worker/testShell.js (hunk 3 above)
Comment (darkwing) [inline tests/service-worker/testShell.js]: Adding back!

File: tests/service-worker/testShell.js (hunk 4 above)
Comment (darkwing) [inline tests/service-worker/testShell.js]: Yeah, we can do that.

File: wp-offline-content/readme.txt (hunk 5 above)
Comment (darkwing) [inline wp-offline-content/readme.txt]: Updated!

**Comment by darkwing** [comment]
All updates made; we should address the CDN and jQuery/enqueue issue after merge.

**Pull request closed without merging**